lib/src/Dscp/Snowdrop/AccountValidation.hs

```diff
@@ -83,86 +84,82 @@ assertSigned WithSignature {..} message = do
+        , HasGetter pk addr
+        , HasGetter proof (WithSignature pk sig (hash, pk))
+        , HasKeyValue id value (AccountId addr) Account
+        , IdStorage stTxId AccountTx
+        , IdSumPrefixed id
+        , Ord addr
+        , Ord id
+        , Eq hash
+        , Eq pk
+        )
+
 validateSimpleMoneyTransfer
-  :: forall e pk addr id proof value ctx sig hash stTxId
-    .  ( HasExceptions e
-          [ AccountValidationException
-          , TxValidationException
-          , StatePException
-          ]
-       , VerifySign pk sig (hash, Author (AccountId addr))
-       , HasPrism proof hash
-       , HasPrism proof pk
-       , HasPrism proof (WithSignature pk sig (hash, Author (AccountId addr)))
-       , HasKeyValue id value (AccountId addr) Account
-       , IdStorage stTxId AccountTx
-       , IdSumPrefixed id
-       , Ord addr
-       , Ord id
-       , Eq hash
-       , HasHash (ChangeSet id value) hash
-       )
-    => Validator e id proof value ctx
+    :: forall       e pk addr id proof value ctx sig hash stTxId
+    .  ValidatorCtx e pk addr id proof value ctx sig hash stTxId
+    => Validator    e         id proof value ctx
+
 validateSimpleMoneyTransfer = mkValidator ty
     [preValidateSimpleMoneyTransfer @e @pk @addr @id @proof @value @ctx @sig @hash @stTxId]
   where
     ty = StateTxType $ getId (Proxy @stTxId) AccountTxId
 
-preValidateSimpleMoneyTransfer
-  :: forall e pk addr id proof value ctx sig hash stTxId
-    .  ( Ord addr
-       , Ord id
-       , HasExceptions e
-          [ AccountValidationException
-          , TxValidationException
-          , StatePException
-          ]
-       , VerifySign pk sig (hash, Author (AccountId addr))
-       , HasPrism proof hash
-       , HasPrism proof pk
-       , HasPrism proof (WithSignature pk sig (hash, Author (AccountId addr)))
-       , HasKeyValue id value (AccountId addr) Account
-       , IdStorage stTxId AccountTx
-       , IdSumPrefixed id
-       , Eq hash
-       , HasHash (ChangeSet id value) hash
-       )
-    => PreValidator e id proof value ctx
-preValidateSimpleMoneyTransfer =
-    PreValidator $ \_trans@StateTx {..} -> do
-        --ty = StateTxType $ getId (Proxy @id) AccountTxId
-        --author :: Author (AccountId addr) <- requirePart txProof (ProofProjectionFailed ty)
-        signedHash
-          :: WithSignature pk sig (hash, Author (AccountId addr))
-          <- requirePart txProof SignatureIsMissing
+authenticate
+    :: forall       e pk addr id proof value ctx sig hash stTxId
+    .  ValidatorCtx e pk addr id proof value ctx sig hash stTxId
+    => Eq             pk
+    => proof
+    -> ERoComp      e         id       value ctx (AccountId addr, Account)
+authenticate proof = do
+    let signedHash :: WithSignature pk sig (hash, pk)
+            = gett proof
 
-        realHashfromExpander
-          :: hash
-          <- requirePart txProof SignatureIsMissing
+        realHashfromExpander :: hash
+            = gett proof
```

Comment:
Why do you wrap lines like this? :thinking:

Reply:
I also do
```
data Foo
    = Bar
    | Qux
```
And in the language I'm doing in my free time, i put `;` before definitions. That's just I.

Reply:
Please try to stick to the suggested coding style though, `hindent` does its job pretty well.